internal/drivers/nomad/driver.go

```diff
@@ -185,39 +185,55 @@ func (p *config) Create(ctx context.Context, opts *types.InstanceCreateOpts) (*t
 	logr.Infoln("scheduler: found a node with available resources")
 
 	// get the machine details where the resource job was allocated
-	ip, id, hostPort, err := p.fetchMachine(logr, resourceJobID)
+	ip, id, liteEngineHostPort, gitspacesPorts, err := p.fetchMachine(logr, resourceJobID)
 	if err != nil {
 		defer p.deregisterJob(logr, resourceJobID, false) //nolint:errcheck
 		return nil, err
 	}
 
+	var gitspacesPortMappingsString string
+	gitspacesPortMappings := make(map[int]int)
+
+	if len(opts.GitspaceOpts.Ports) != len(gitspacesPorts) {
+		return nil, fmt.Errorf("scheduler: could not allocate required number of ports for gitspaces")
+	} else {
```

Comment:
this will run for non git space ?

Reply:
For non-gitspace, both `opts.GitspaceOpts.Ports` and `gitspacesPorts` will be nil. 
Nil slices have length of 0, so it will go to the else block.
In the else block, since the length of opts.GitspaceOpts.Ports is 0, it will not run the for loop